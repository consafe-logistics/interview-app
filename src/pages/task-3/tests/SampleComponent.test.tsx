import { render, screen } from "@testing-library/react";
import SampleComponent from "./SampleComponent";

describe('SampleComponent', () => {
  it('should render correctly', () => {
    const tree = render(<SampleComponent />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it('should increment the count to 3 after 3 clicks', () => {
    // Setup
    render(<SampleComponent />);

    // Act
    const button = screen.getByLabelText('Increment');
    button.click();
    button.click();
    button.click();

    // Expect
    const label = screen.getByDisplayValue('Clicks: 3');
    expect(label.innerText).toContain(3)
  });

  it('should open the modal after button click', () => {
    // Setup
    const tree = render(<SampleComponent />).asFragment();

    // Act
    const button = screen.getByText('Open Modal');
    button.click();

    // Expect
    const modal = screen.getByRole('dialog');
    expect(tree).toContain(modal);
  });
});