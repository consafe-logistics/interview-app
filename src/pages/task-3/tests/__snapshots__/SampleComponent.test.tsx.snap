// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SampleComponent should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Clicks: 0
    </p>
    <button
      class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Increment
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
